controllers/module_reconciler.go

```diff
@@ -262,6 +263,24 @@ func (r *ModuleReconciler) handleDevicePlugin(ctx context.Context, mod *ootov1al
 	return err
 }
 
+func (r *ModuleReconciler) getRequestedModule(ctx context.Context, namespacedName types.NamespacedName) (*ootov1alpha1.Module, error) {
+	mods := ootov1alpha1.ModuleList{}
+
+	err := r.Client.List(context.Background(), &mods)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list modules in the cluster: %w", err)
+	}
+
+	r.metrics.SetExistingKMMOsCRs(len(mods.Items))
```

Comment:
Wondering if it is the right place for this call. `getRequestedModule` sounds like a utility function; although it's cheap, updating the metric is a side-effect.

Reply:
i prefered to put it there , so as not to return the number of KMMO in the cluster, because it also sound as a side-effect to actually finding the right module

Reply:
fixed